rs/crypto/secp256k1/src/lib.rs

```diff
@@ -330,6 +340,31 @@ fn pem_encode(raw: &[u8], label: &'static str) -> String {
     })
 }
 
+fn bip341_hash(pk_x: &[u8], ttr: &[u8]) -> Result<Scalar, InvalidTaprootHash> {
```

Comment:
It could be useful to add descriptions for the added functions, e.g., what `ttr` is or that a BIP341 hash is actually the tweak. Maybe we should actually call it `bip341_tweak`. We could also add the assumptions that `pk_x` and `ttr` have a valid length or add `debug_assert`s.

Reply:
Added some comments and length checks